src/AzureAIProxy/Middleware/OpenAIErrorResponse.cs

```diff
@@ -0,0 +1,32 @@
+using System.Net;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace AzureAIProxy.Middleware;
+
+public class OpenAIErrorResponse(string value, HttpStatusCode statusCode)
+{
+    private readonly JsonHttpResult<OpenAIErrorPayload> innerResult = TypedResults.Json(
+        new OpenAIErrorPayload((int)statusCode, value),
+        statusCode: (int)statusCode
+    );
```

Comment:
Is there a point to doing this and not just using the value/status code? In `WriteAsync` we just read those as properties of `innerResult` so why don't we just use the `value` and `statusCode` variables?